src/packagedcode/utils.py

```diff
@@ -215,3 +217,57 @@ def yield_dependencies_from_package_resource(resource, package_uid=None):
     for pkg_data in resource.package_data:
         pkg_data = models.PackageData.from_dict(pkg_data)
         yield from yield_dependencies_from_package_data(pkg_data, resource.path, package_uid)
+
+
+def update_dependencies_as_resolved(dependencies):
+    """
+    For a list of dependency mappings with their respective
+    resolved packages, update in place the dependencies for those
+    resolved packages as resolved (update `is_resolved` as True),
+    if the requirement is also present as a resolved package.
+    """
+    #TODO: Use vers to mark update `is_resolved` even in the case
+    # of incomplete resolution/partially pinned dependencies
+
+    # These are only type, namespace and name (without version and qualifiers)
+    base_resolved_purls = []
+    base_purl_fields = ["type", "namespace", "name"]
+    try:
+        resolved_packages = [
+            dep.get("resolved_package")
+            for dep in dependencies
+            if dep.get("resolved_package")
+        ]
+    except AttributeError:
+        raise Exception(dependencies)
+
+    # No resolved packages are present for dependencies
+    if not resolved_packages:
+        return
+
+    for pkg in resolved_packages:
+        purl_mapping = PackageURL.from_string(purl=pkg.get("purl")).to_dict()
```

Comment:
May be this is best to have as a function to `as_versionless_purl` or something?